docs/architecture/api-20201231-preview.md

```diff
@@ -0,0 +1,148 @@
+# API 20201231-preview design
+
+For api `20201231-preview` bellow is the list of proposed changes and features.
+These would have to land into API and backend as a features.
+
+Feature work will need to be agreed and converted to working stories. All technical details bellow are overview only
+
+## API changes
+
+### ComputeSecurityProfile
+
+Add security profile to worker and master pools configuration for all security enhancements
+for the compute instances
+
+We need to make sure that these option are validated on the cluster too when customer is interacting with MachineSet objects. We should be able to verify if these options where enabled on cluster create and set those accordingly. This does not prevent customer to create instances with and without encryption in the same cluster.
+
+```
+// MasterProfile represents a master profile.
+type MasterProfile struct {
+	...
+
+	ComputeSecurityProfile
```

Comment:
I think it's going to be more effort/irritation to have the shared anonymous struct than not to.  I wouldn't bother.
* Master/Worker fields already mostly overlap but we don't use a shared struct
* Anonymous structs are a pain to initialise in Golang clients
* Don't know if our swagger tooling handles anonymous structs
* Don't know what the knock-on effects are in other language bindings